design-a/contracts/Valset.sol

```diff
@@ -0,0 +1,57 @@
+pragma solidity ^0.4.11;
+
+contract Valset {
+    struct Validator {
+        address Address;
+        uint64 Power;
+    }
+
+    Validator[] public validators;
+    uint64 public totalPower;
+
+    uint updateSeq = 0;
+
+    function verify(bytes32 hash, uint8[] v, bytes32[] r, bytes32[] s) returns (bool) {
+        if (!(v.length == r.length &&
+              r.length == s.length)) {
+            return false;
+        }
+
+        uint64 signedPower = 0;
+
+        for (uint i = 0; i < v.length; i++) {
```

Comment:
Here is what i don't like about this algorithm, let's says that signatures for val 1, 3, 5 are enough voting power to be > 2/3rd's of the voting power.

I still need to submit signatures for 2 and 4 too.

Consider submitting the indexes of validators in the arguments so we only need to submit the minimum number of signatures.